### PR TITLE
implement progressive web app stuff and socket auto-reconnect

### PR DESCRIPTION
**Pull request by 5harf**
:)

### CODE REVIEW

File: dist/client/app/allDishes/allDishes.html [hunk 1]
```diff
@@ -1 +1,87 @@
...
\ No newline at end of file
+<!-- PAGINATION -->
+<ul class="pagination">
```
Comment (mi-lee) [inline dist/client/app/allDishes/allDishes.html:2]: Shouldn't this be minified because it's in `dist/`?

**Comment by mi-lee** [comment]
R+ other than the `dist/` folders needing to be minified

File: dist/client/app/allDishes/allDishes.html (hunk 1 above)
Comment (5harf) [inline dist/client/app/allDishes/allDishes.html:2]: The htmlmin grunt task was doing something weird and the app wouldn't load with it on so I took it out...  Not sure why.  You could try setting it up to work again.

**Comment by 5harf** [comment]
What does R+ mean?

**Comment by mi-lee** [comment]
haha means you're free to merge the PR. my boss used to work at mozilla https://wiki.mozilla.org/Firefox/Code_Review#What_does_r.2B_mean.3F

File: dist/client/app/allDishes/allDishes.html (hunk 1 above)
Comment (mi-lee) [inline dist/client/app/allDishes/allDishes.html:2]: does our heroku link use the dist/ anyway? I can't remember.

I think if you do each Grunt task individually it should work?

**Comment by 5harf** [comment]
damn working with the best over there :O

**Comment by mi-lee** [comment]
you too 😲

**Pull request merged**